utils/src/utils.cpp

```diff
@@ -1,44 +1,78 @@
 #include "utils.hpp"
 
+#include <chrono>
 #include <vector>
 
 namespace util {
 
+uint64_t get_time_ms() {
+    const auto timestamp = std::chrono::system_clock::now();
+    return std::chrono::duration_cast<std::chrono::milliseconds>(
+               timestamp.time_since_epoch())
+        .count();
+}
+
 constexpr uint8_t hex_to_nibble(const char & ch)
 {
-  return ( ch >= '0' && ch <= '9') ? ch - 48 : ((ch >= 'A' && ch <= 'F' ) ? ch - 55 : ((ch >= 'a' && ch <= 'f' ) ? ch - 87 : 0));
+    return ( ch >= '0' && ch <= '9') ? ch - 48 : ((ch >= 'A' && ch <= 'F' ) ? ch - 55 : ((ch >= 'a' && ch <= 'f' ) ? ch - 87 : 0));
 }
 
 constexpr uint8_t hexpair_to_byte(const char & hi, const char & lo)
 {
-  return hex_to_nibble(hi) << 4 | hex_to_nibble(lo);
+    return hex_to_nibble(hi) << 4 | hex_to_nibble(lo);
 }
 
 std::string hex64_to_base32z(const std::string &src)
 {
-  // decode to binary
-  std::vector<uint8_t> bin;
-  // odd sized is invalid
-  if(src.size() & 1)
-    return "";
-  {
-    auto itr = src.begin();
-    while(itr != src.end())
+    // decode to binary
+    std::vector<uint8_t> bin;
+    // odd sized is invalid
+    if(src.size() & 1)
+        return "";
     {
-      const char hi = *itr;
-      ++itr;
-      const char lo = *itr;
-      ++itr;
-      bin.emplace_back(hexpair_to_byte(hi,lo));
+        auto itr = src.begin();
+        while(itr != src.end())
+        {
+            const char hi = *itr;
+            ++itr;
+            const char lo = *itr;
+            ++itr;
+            bin.emplace_back(hexpair_to_byte(hi,lo));
+        }
+    }
+    // encode to base32z
+    char buf[64] = {0};
+    std::string result;
+    if (char const *dest = base32z_encode(bin, buf))
+        result = dest;
+
+    return result;
+}
+
+bool parseTimestamp(const std::string& timestampString, const uint64_t& ttl,
```

Comment:
Better to pass ttl as plain `uint64_t`, i.e. by value.

Reply:
Roger, fixing